framework/include/gpu/base/GPUDatum.h

```diff
@@ -0,0 +1,180 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "GPUTypes.h"
+#include "GPUAssembly.h"
+#include "GPUSystem.h"
+#include "GPUVariable.h"
+
+class Datum
```

Comment:
Docstrings. Move function definitions outside of class declaration. Needs to be namespaced